app/controllers/concerns/validators_controller_helper.rb

```diff
@@ -6,17 +6,19 @@ def create_json_result(validator, with_history = false)
 
     hash.merge!(validator.to_builder.attributes!)
 
-    score = validator.score
-    ip = score.ip_for_api if score
-    vote_account = validator.vote_accounts.last
+    validator_score = with_history ? validator.validator_score_v1 : validator.validator_score_v1_for_api
+
+    data_center = validator.validator_ip_active_for_api&.data_center_host_for_api&.data_center_for_api
```

Comment:
This works (does not trigger n+1 queries) only if chained models are included in sql query, so I think we could leave it for now only here since query is only used here. You can use `validator.data_center_for_api` when need data_center for single validator.